stdlib/list.ml

```diff
@@ -289,6 +289,33 @@ and[@tail_mod_cons] prepend_concat_map ys f xs =
   | [] -> concat_map f xs
   | y :: ys -> y :: prepend_concat_map ys f xs
 
+let take n l =
+  let rec aux i acc = function
+    | x::l when i < n -> aux (i + 1) (x::acc) l
+    | _rest -> rev acc
+  in
+  if n < 0 then invalid_arg "List.take";
+  aux 0 [] l
```

Comment:
```ocaml
  let take n l =
    if n < 0 then invalid_arg "List.take"
    else
      let[@tail_mod_cons] rec loop n l =
        if n = 0 then []
        else
          match l with
          | [] -> []
          | x::l -> x::loop (n - 1) l
      in loop n l
```

One option for a slightly more performant version with `@tail_mod_cons`.

Reply:
I used a slightly more concise version instead as i felt this version could be improved visually a little (less vertical space), could you check that it behave the same performance wise? I also changed `take_while` to use tail_mod_cons.

Reply:
Yep the new version seems to have the same runtime and heap allocation behaviour. Very elegant way to reduce the vertical space, nice work! 
![image](https://github.com/ocaml/ocaml/assets/28349560/298f0b6f-379a-4a96-a4cd-7af18f418355)
![image](https://github.com/ocaml/ocaml/assets/28349560/f9ec89c8-e63b-4757-b0b5-211a4d138d5b)